File: Fantasy_Football/ClientApp/src/app/models/player-detail.ts


    export interface PlayerDetail {
        season:      number;
        projections: Projections;
    }
    
    export interface Projections {
        QB:  Qb[];
        RB:  Rb[];
        WR:  Rb[];
        TE:  Rb[];
        K:   K[];
        DEF: Def[];
        IDP: Idp[];
    }
    
    export interface Def {
        playerId:             string;
        name:                 string;
        team:                 string;
        position:             DEFPosition;
        fumbled_forced:       string;
        sacks:                string;
        defensive_touchdowns: string;
        safeties:             string;
        points_allowed:       string;
        yards_allowed:        string;
        proj_pts:             string;
    }
    
    export enum DEFPosition {
        Def = "DEF",
    }
    
    export interface Idp {
        playerId:                 string;
        name:                     string;
        team:                     string;
        position:                 IDPPosition;
        tackles:                  string;
        assists:                  string;
        sacks:                    string;
        passes_defended:          string;
        fumbled_forced:           string;
        interceptions:            string;
        interception_touchdowns:  string;
        fumble_return_touchdowns: string;
        proj_pts:                 string;
    }
    
    export enum IDPPosition {
        Cb = "CB",
        De = "DE",
        Dt = "DT",
        LB = "LB",
        S = "S",
    }
    
    export interface K {
        playerId:              string;
        name:                  string;
        team:                  string;
        position:              KPosition;
        field_goals_made:      string;
        field_goals_attempted: string;
        extra_points_made:     string;
        proj_pts:              string;
    }
    
    export enum KPosition {
        K = "K",
    }
    
    export interface Qb {
        playerId:              string;
        name:                  string;
        team:                  string;
        position:              QBPosition;
        passing_attempts:      string;
        passing_completions:   string;
        passing_yards:         string;
        passing_touchdowns:    string;
        passing_interceptions: string;
        rushing_attempts:      string;
        rushing_yards:         string;
        rushing_touchdowns:    string;
        proj_pts:              string;
    }
    
    export enum QBPosition {
        Qb = "QB",
    }
    
    export interface Rb {
        playerId:             string;
        name:                 string;
        team:                 string;
        position:             RBPosition;
        rushing_attempts:     string;
        rushing_yards:        string;
        rushing_touchdowns:   string;
        fumbles:              string;
        receptions:           string;
        receiving_yards:      string;
        receiving_touchdowns: string;
        targets:              string;
        proj_pts:             string;
    }
    
    export enum RBPosition {
        Rb = "RB",
        Te = "TE",
        Wr = "WR",
    }
    

File: Fantasy_Football/Models/PlayerDetail.cs
namespace Fantasy_Football.Models
{
   

        public class PlayerDetail
        {
            public int season { get; set; }
            public Projections projections { get; set; }
        }

        public class Projections 
        {
            public QB[] QB { get; set; }
            public RB[] RB { get; set; }
            public WR[] WR { get; set; }
            public TE[] TE { get; set; }
            public K[] K { get; set; }
            public DEF[] DEF { get; set; }
            public IDP[] IDP { get; set; }
        }

        public class QB
        {
            public string playerId { get; set; }
            public string name { get; set; }
            public string team { get; set; }
            public string position { get; set; }
            public string passing_attempts { get; set; }
            public string passing_completions { get; set; }
            public string passing_yards { get; set; }
            public string passing_touchdowns { get; set; }
            public string passing_interceptions { get; set; }
            public string rushing_attempts { get; set; }
            public string rushing_yards { get; set; }
            public string rushing_touchdowns { get; set; }
            public string proj_pts { get; set; }
        }

        public class RB
        {
            public string playerId { get; set; }
            public string name { get; set; }
            public string team { get; set; }
            public string position { get; set; }
            public string rushing_attempts { get; set; }
            public string rushing_yards { get; set; }
            public string rushing_touchdowns { get; set; }
            public string fumbles { get; set; }
            public string receptions { get; set; }
            public string receiving_yards { get; set; }
            public string receiving_touchdowns { get; set; }
            public string targets { get; set; }
            public string proj_pts { get; set; }
        }

        public class WR
        {
            public string playerId { get; set; }
            public string name { get; set; }
            public string team { get; set; }
            public string position { get; set; }
            public string rushing_attempts { get; set; }
            public string rushing_yards { get; set; }
            public string rushing_touchdowns { get; set; }
            public string fumbles { get; set; }
            public string receptions { get; set; }
            public string receiving_yards { get; set; }
            public string receiving_touchdowns { get; set; }
            public string targets { get; set; }
            public string proj_pts { get; set; }
        }

        public class TE
        {
            public string playerId { get; set; }
            public string name { get; set; }
            public string team { get; set; }
            public string position { get; set; }
            public string rushing_attempts { get; set; }
            public string rushing_yards { get; set; }
            public string rushing_touchdowns { get; set; }
            public string fumbles { get; set; }
            public string receptions { get; set; }
            public string receiving_yards { get; set; }
            public string receiving_touchdowns { get; set; }
            public string targets { get; set; }
            public string proj_pts { get; set; }
        }

        public class K
        {
            public string playerId { get; set; }
            public string name { get; set; }
            public string team { get; set; }
            public string position { get; set; }
            public string field_goals_made { get; set; }
            public string field_goals_attempted { get; set; }
            public string extra_points_made { get; set; }
            public string proj_pts { get; set; }
        }

        public class DEF
        {
            public string playerId { get; set; }
            public string name { get; set; }
            public string team { get; set; }
            public string position { get; set; }
            public string fumbled_forced { get; set; }
            public string sacks { get; set; }
            public string defensive_touchdowns { get; set; }
            public string safeties { get; set; }
            public string points_allowed { get; set; }
            public string yards_allowed { get; set; }
            public string proj_pts { get; set; }
        }

        public class IDP
        {
            public string playerId { get; set; }
            public string name { get; set; }
            public string team { get; set; }
            public string position { get; set; }
            public string tackles { get; set; }
            public string assists { get; set; }
            public string sacks { get; set; }
            public string passes_defended { get; set; }
            public string fumbled_forced { get; set; }
            public string interceptions { get; set; }
            public string interception_touchdowns { get; set; }
            public string fumble_return_touchdowns { get; set; }
            public string proj_pts { get; set; }
        }


    
}
